src/components/Register/Register.js

```diff
@@ -0,0 +1,111 @@
+import React, { Component } from 'react';
+
+class Register extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      email: '',
+      password: '',
+      name: ''
+    }
+  }
+
+  onChangeName = (event) => {
+    this.setState({ name: event.target.value })
+  }
+
+  onEmailChange = (event) => {
+    this.setState({ email: event.target.value })
+  }
+
+  onPasswordChange = (event) => {
+    this.setState({ password: event.target.value })
+  }
+
+  onSubmitSignIn = () => {
```

Comment:
When handle forms with JS asynchronously don't forget to prevent default HTML form submit actions. Read more https://www.robinwieruch.de/react-preventdefault

```suggestion
  onSubmitSignIn = event => {
    event.preventDefault();
```